db/etcd.go

```diff
@@ -569,259 +406,177 @@ func (etcddb etcdDB) InitTestRun(testUuid string, testAgentMap map[string]map[st
+		fmt.Sprintf("/todd/testruns/%s/agents/%s/status", testUUID, agentUUID), // key
 		status, // value
 		nil,    //optional args
 	)
 	if err != nil {
-		log.Errorf("Problem updating status for agent %s in test %s", agentUuid, testUuid)
+		log.Errorf("Problem updating status for agent %s in test %s", agentUUID, testUUID)
 		log.Error(err)
-		os.Exit(1)
+		return err
 	}
 
 	return nil
 }
 
 // SetAgentTestData sets the post-test data for an agent in a particular testrun
-func (etcddb etcdDB) SetAgentTestData(testUuid, agentUuid, testData string) error {
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
-
-	_, err = kapi.Set(
+func (etcddb *etcdDB) SetAgentTestData(testUUID, agentUUID, testData string) error {
+	_, err := etcddb.keysAPI.Set(
 		context.Background(),                                                     // context
-		fmt.Sprintf("/todd/testruns/%s/agents/%s/testdata", testUuid, agentUuid), // key
+		fmt.Sprintf("/todd/testruns/%s/agents/%s/testdata", testUUID, agentUUID), // key
 		testData, // value
 		nil,      //optional args
 	)
 	if err != nil {
-		log.Errorf("Problem updating testdata for agent %s in test %s", agentUuid, testUuid)
+		log.Errorf("Problem updating testdata for agent %s in test %s", agentUUID, testUUID)
 		log.Error(err)
-		os.Exit(1)
+		return err
 	}
 
 	return nil
 }
 
 // GetTestStatus returns a map containing a list of agent UUIDs that are participating in the provided test, and their status in this test.
-func (etcddb etcdDB) GetTestStatus(testUuid string) map[string]string {
+func (etcddb *etcdDB) GetTestStatus(testUUID string) (map[string]string, error) {
 
-	ret_map := make(map[string]string)
+	retMap := make(map[string]string)
 
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
+	keyStr := fmt.Sprintf("/todd/testruns/%s/agents", testUUID)
 
-	key_str := fmt.Sprintf("/todd/testruns/%s/agents", testUuid)
+	log.Debug("Retrieving detailed test status for ", testUUID)
 
-	log.Debug("Retrieving detailed test status for ", testUuid)
-
-	resp, err := kapi.Get(context.Background(), key_str, &client.GetOptions{Recursive: true})
+	resp, err := etcddb.keysAPI.Get(context.Background(), keyStr, &client.GetOptions{Recursive: true})
 	if err != nil {
 		fmt.Println(err)
```

Comment:
Eww, this should be a log statement. This was my bad, but do you mind adding a log statement in your patch?